docs: IA calibration (P0): add Tune "Noise & relevance" and "Release timing" pages (#705)
Promptless/docs: src/content/docs/docs/tune/noise-and-relevance-filtering.mdx

Reviewer: There's no specific information about tuning relevance via the PROMPTLESS.md and Knowledge Base. For example, when passively listening to a Slack channel, the easiest way to tune for which messages to act on is to add guidelines in the Knowledge Base for what to act on, what not to act on, and how to handle edge cases or specific processes.

Reviewer: We should be referring to UI strings, not YAML configs.

---
title: Filter noise and tune relevance
description: >-
  Reduce low-value documentation suggestions by scoping what triggers Promptless
  and teaching it which changes to skip
slug: docs/tune/noise-and-relevance-filtering
sidebar:
  hidden: false
  order: 22
  label: Noise & relevance
---
import Note from '@components/fern/Note.astro';
import Tip from '@components/fern/Tip.astro';
import Info from '@components/fern/Info.astro';

> The first few days decide whether you keep trusting Promptless. If it suggests documentation for every internal refactor and config bump, you tune it out; if it stays quiet on the changes that matter, you stop relying on it. Tuning for relevance means narrowing what reaches Promptless in the first place and teaching it, over time, which changes are worth documenting.

Reviewer: ```suggestion
The first few days of use decide whether you keep trusting Promptless. If your get suggestions for every internal refactor and config bump, you tune it out. If Promptless stays quiet on the changes that matter, you stop relying on it. Tuning for relevance means narrowing what reaches Promptless in the first place and teaching it, over time, which changes are worth documenting.
```

Promptless filters at three points, from earliest to latest:

1. **What triggers analysis** — scope your triggers so only documentation-worthy events reach Promptless.
2. **What Promptless skips automatically** — built-in filters drop changes that are never documentation-worthy.

> 3. **What you teach it to skip** — feedback on suggestions durably shifts what Promptless proposes next time.

Reviewer: ```suggestion
3. **What you teach it to skip** — feedback on suggestions changes what Promptless proposes next time.
```

```yaml
triggers:
  tagged-repos:
    trigger_type: github_pr
    match:
      - repos: all
        excluded_repos:
          - acme/infra-scripts
        repo_topics:
          - docs-watch
        trigger_on:
          - opened
```

### Trigger later in the release cycle

The event you trigger on decides how often Promptless runs and how finished a change is when it does. `opened` fires on every new pull request; `first_approval` waits until a reviewer has vetted the change; `merge` waits until it ships. Triggering later means fewer, higher-signal runs. [Tune what Promptless suggests → Release timing](/docs/tune/release-timing) covers how to match the trigger event to your release stage.

<Info>
The `if` clause in [publishing policies](/docs/reference/configuration-reference#policies) can change how a suggestion is published — where it notifies, whether it opens a PR — but it cannot stop a suggestion from being created. To keep a change from becoming a suggestion at all, scope the trigger that would fire on it.
</Info>

## What Promptless skips automatically

> Some changes are never documentation-worthy, so Promptless drops them before it creates a suggestion. You don't configure these filters; they always apply.

Reviewer: ```suggestion
Some changes are never documentation-worthy, so Promptless drops them before it creates a suggestion. You don't configure these filters because they always apply.
```

When Promptless evaluates a change and decides no documentation is needed, that decision appears on the Triggers page in your dashboard, so a missing suggestion reads as a considered "no" rather than a silent miss. See [The web interface](/docs/work-the-queue/web-interface) for where to find it.
</Note>

## Teach Promptless what to skip

Trigger scope handles the coarse cuts. For the judgment calls — a class of change that technically touches docs but that your team never documents — teach Promptless through feedback. When you save feedback on a suggestion, it resolves the suggestion in front of you and becomes a durable convention Promptless applies to future ones.

### Close a suggestion with a reason

When a suggestion isn't worth keeping, close it and pick the reason that fits. The options include several that map directly to noise:

- This change is too insignificant
- This change is self-explanatory and doesn't need docs
- This change was covered in another Promptless suggestion
- This is already in my docs

Check **Remember this feedback for future suggestions** so the reason shapes what Promptless proposes next time, not just this once.

### Remember a durable preference

> Any feedback you give — closing a suggestion, requesting changes, or leaving a comment — can be saved for the future. Phrase it as a standing rule so Promptless generalizes it:

Reviewer: ```suggestion
Any feedback you give — like closing a suggestion, requesting changes, or leaving a comment — can be saved for the future. Phrase it as a standing rule so Promptless generalizes it:
```

- "Don't create suggestions for dependency bumps."
- "Internal-only tooling changes don't need docs."

Promptless records what it learns in your [Agent Knowledge Base](/docs/get-the-most-out/agent-knowledge-base), so the preference persists across every future suggestion. The full set of feedback channels is documented in [Teach conventions with feedback](/docs/tune/teaching-conventions).

<Tip>

> Filtering compounds. Trigger scope removes whole categories of change up front; feedback refines the judgment calls that slip through. Revisit both after your first week — the suggestions you keep closing point to the next trigger or convention to add.

Reviewer: ```suggestion
Filtering compounds. Trigger scope removes whole categories of change up front, while feedback refines the judgment calls that slip through. Revisit both after your first week — the suggestions you keep closing point to the next trigger or convention to add.
```
